sklearn/linear_model/_logistic.py

```diff
@@ -2016,15 +2028,54 @@ def score(self, X, y, sample_weight=None):
         sample_weight : array-like of shape (n_samples,), default=None
             Sample weights.
 
+        **score_params : dict
+            Parameters to pass to the `score` method of the underlying scorer.
+
         Returns
         -------
         score : float
             Score of self.predict(X) wrt. y.
         """
-        scoring = self.scoring or "accuracy"
-        scoring = get_scorer(scoring)
+        scoring = self._get_scorer()
+        routed_params = process_routing(
+            obj=self,
+            method="score",
+            sample_weight=sample_weight,
+            other_params=score_params,
+        )
+
+        return scoring(
+            self, X, y, sample_weight=sample_weight, **routed_params.scorer.score
```

Comment:
`routed_params.scorer.score` would include `sample_weight`, no need to pass it explicitly.